test/date.js

```diff
@@ -153,15 +153,25 @@ describe('Date Component', () => {
           'date-field': error
         };
         date.hooks['pre-render'](req, res, next);
-        expect(res.render).to.have.been.calledWith(path.resolve(__dirname, '../templates/date.html'), { error, key: 'date-field', legend: null, hint: null });
+        expect(res.render).to.have.been.calledWith(path.resolve(__dirname, '../templates/date.html'), { error, key: 'date-field', legend: null, legendClassName: '', hint: null });
       });
 
-      it('calls res.render with template, key, legend and hint with values NOT null', () => {
+      it('calls res.render with template, key, legend and hint with values NOT null OR empy string', () => {
         req.translate = sinon.stub();
         req.translate.withArgs('fields.date-field.legend').returns('you legend');
         req.translate.withArgs('fields.date-field.hint').returns('some hint');
         date.hooks['pre-render'](req, res, next);
-        expect(res.render).to.have.been.calledWith(path.resolve(__dirname, '../templates/date.html'), { error: undefined, key: 'date-field', legend: 'you legend', hint: 'some hint' });
+        expect(res.render).to.have.been.calledWith(path.resolve(__dirname, '../templates/date.html'), { error: undefined, key: 'date-field', legendClassName: '', legend: 'you legend', hint: 'some hint' });
+      });
+
+      it('calls res.render with template, key, legend and hint with values NOT null OR empy string', () => {
+        date = dateComponent('date-field', {
+          legend: {
+            className: 'testClass'
+          }
+        });
+        date.hooks['pre-render'](req, res, next);
+        expect(res.render).to.have.been.calledWith(path.resolve(__dirname, '../templates/date.html'), { error: undefined, key: 'date-field', legendClassName: 'testClass', legend: null, hint: null });
```

Comment:
I'd try to avoid testing things which you don't care about as it results in fragile tests when unrelated things change. 

It looks like the only thing we actually want to test here is that the `legendClassName` property is visible, so I'd swap the expect for something like:

```js
const args = res.render.lastCall.args;
expect(args[1]).to.have.property('legendClassName');
expect(args[1].legendClassName).to.equal('testClass');
```

This approach makes the tests a lot easier to debug if they fail too as it means you don't have to work out exactly which of the large number of implicit assertions are failing.